src/components/pages/user/components/subscription-details.tsx

```diff
@@ -0,0 +1,133 @@
+
+type SubscriptionDetailsProps = {
+  stripeCustomerId: string
+  slug: string
+}
+
+const SubscriptionDetails: React.FunctionComponent<SubscriptionDetailsProps> =
+  ({stripeCustomerId, slug}) => {
+    const {viewer} = useViewer()
+    const {subscriptionData, loading} = useSubscriptionDetails({
+      stripeCustomerId,
+    })
+
+    const subscriptionName = subscriptionData && subscriptionData.product?.name
+    const subscriptionUnitAmount = get(
+      subscriptionData,
+      'latestInvoice.amount_due',
+      subscriptionData?.price?.unit_amount,
+    )
+    const currency = get(
+      subscriptionData,
+      'latestInvoice.currency',
+      subscriptionData?.price?.unit_amount,
+    )
+    const subscriptionPrice =
+      subscriptionUnitAmount &&
+      currency &&
+      new Intl.NumberFormat('en-US', {
+        style: 'currency',
+        currency: currency,
+        minimumFractionDigits: 0,
+      }).format(subscriptionUnitAmount / 100)
+
+    return !loading && subscriptionData ? (
+      <div className="w-full">
+        {subscriptionName ? (
+          <>
+            <h3 className="mb-2 text-lg font-medium">
+              ⭐️ You're an <strong>egghead Member!</strong>
+            </h3>
+            <p className="text-accents-5">
+              You can update your plan and payment information below via Stripe.
+            </p>
+            <div className="mt-8 mb-4 font-semibold">
+              {!subscriptionData?.portalUrl ? (
+                <div className="h-12 mb-6">loading</div>
+              ) : subscriptionPrice ? (
+                <div className="flex space-x-2">
+                  <div>
+                    You are currently paying{' '}
+                    {`${subscriptionPrice}/${recur(subscriptionData.price)}`}{' '}
+                    for your membership
+                  </div>
+                  {subscriptionData?.subscription?.cancel_at_period_end && (
+                    <div className="flex items-center justify-center px-2 py-1 text-xs bg-gray-100 rounded text-gray-1000 mt-2">
+                      cancelled
+                    </div>
+                  )}
+                </div>
+              ) : (
+                <Link href="/pricing">
+                  <a
+                    onClick={() => {
+                      track(`clicked pricing`, {
+                        location: 'accounts',
+                      })
+                    }}
+                  >
+                    Join today!
+                  </a>
+                </Link>
+              )}
+            </div>
+          </>
+        ) : (
+          <>
+            <h3 className="mb-2 text-lg font-medium">
+              No paid subscription found.
+            </h3>
+            {(viewer.is_pro || viewer.is_instructor) && (
+              <p>
+                You still have access to a Pro Membership. If you feel this is
+                in error please email{' '}
+                <a
```

Comment:
I know there are some weird scenarios that lead to a person having access. But I wonder if one of the main scenarios in which this message displays is because a person has access through team membership. In which case, the messaging might be confusing.

Reply:
On second thought, I think this component is only going to render if the user has (owned) accounts tied to them. Which is to say, that only account owners are going to see this. My previous comment, in that case, is not a concern.